pyabc2/key.py

```diff
@@ -185,7 +185,8 @@ def _scale_intervals(
         Whether to return 7 intervals by computing the interval from scale degree 7 to 8,
         by default True.
     """
-    assert len(values) == 7
+    if len(values) != 7:
+        raise ValueError(f"scales should contain 7 intervals, not {len(values)}")
```

Comment:
The inputs here are relative (to the first, such that the first value is 0) values in half-step space. Maybe something like this:
```suggestion
        raise ValueError(f"expected 7 values, got {len(values)}")
```
Maybe also put in the check that first value is 0 and last is < 12